### PR TITLE
instantsend: Refactor various parts

### PR DESCRIPTION
**Pull request by UdjinM6**
Please see individual commits.

Extracted from https://github.com/dashpay/dash/pull/3970

### CODE REVIEW

File: src/llmq/quorums_instantsend.cpp [hunk 1]
```diff
@@ -1040,7 +996,37 @@ void CInstantSendManager::BlockConnected(const std::shared_ptr<const CBlock>& pb
...
+                db.WriteInstantSendLockMined(islockHash, pindex->nHeight);
+            }
+        }
+
+        if (!masternodeSync.IsBlockchainSynced()) {
+            continue;
+        }
+
+        bool non_locked = islockHash.IsNull() && !chainLocksHandler->HasChainLock(pindex->nHeight, pindex->GetBlockHash());
+        if (non_locked) {
```
Comment (xdustinface) [inline src/llmq/quorums_instantsend.cpp:1020]:
```suggestion
        if (islockHash.IsNull() && !chainLocksHandler->HasChainLock(pindex->nHeight, pindex->GetBlockHash())) {
```

**Review by xdustinface — APPROVED** [review]
Looks good, utACK with one maybe nit :)

**Review by xdustinface — DISMISSED** [review]
utACK

File: src/llmq/quorums_instantsend.cpp [hunk 2]
```diff
@@ -963,62 +963,29 @@ void CInstantSendManager::ProcessInstantSendLock(NodeId from, const uint256& has
...
-
 void CInstantSendManager::TransactionAddedToMempool(const CTransactionRef& tx)
 {
-    if (!IsInstantSendEnabled()) {
+    if (!IsInstantSendEnabled() || !masternodeSync.IsBlockchainSynced() || tx->vin.empty()) {
         return;
     }
 
-    CInstantSendLockPtr islock{nullptr};
+    CInstantSendLockPtr islock;
```
Comment (PastaPastaPasta) [inline src/llmq/quorums_instantsend.cpp:961]: why remove the `{nullptr}`?

Comment (UdjinM6) [reply src/llmq/quorums_instantsend.cpp:961]: It's assigned right below unconditionally, so there is no need to initialize it here and it should be safe. But it wasn't intentional actually, was just smth I did while moving things around. Let's revert this.

```suggestion
    CInstantSendLockPtr islock{nullptr};
```

**Review by xdustinface — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Pull request merged**